packages/headless/src/controllers/commerce/core/facets/category/headless-commerce-category-facet-search.ts

```diff
@@ -0,0 +1,56 @@
+import {CategoryFacetSearchResult} from '../../../../../api/search/facet-search/category-facet-search/category-facet-search-response';
+import {CommerceEngine} from '../../../../../app/commerce-engine/commerce-engine';
+import {
+  executeCommerceFacetSearch,
+  executeCommerceFieldSuggest,
+} from '../../../../../features/commerce/facets/facet-search-set/commerce-facet-search-actions';
+import {categoryFacetSearchSetReducer as categoryFacetSearchSet} from '../../../../../features/facets/facet-search-set/category/category-facet-search-set-slice';
+import {CategoryFacetSearchSection} from '../../../../../state/state-sections';
+import {loadReducerError} from '../../../../../utils/errors';
+import {
+  CategoryFacetSearchProps as CoreCategoryFacetSearchProps,
+  buildCoreCategoryFacetSearch,
+} from '../../../../core/facets/facet-search/category/headless-category-facet-search';
+import {CoreFacetSearchState} from '../searchable/headless-commerce-searchable-facet';
+
+export type CategoryFacetSearchProps = Omit<
+  CoreCategoryFacetSearchProps,
+  'executeFacetSearchActionCreator' | 'executeFieldSuggestActionCreator'
+>;
+
+export type CategoryFacetSearchState =
+  CoreFacetSearchState<CategoryFacetSearchResult>;
+
+export type CategoryFacetSearch = Omit<
+  ReturnType<typeof buildCoreCategoryFacetSearch>,
+  'showMoreResults' | 'updateCaptions' | 'state'
+> & {
+  state: CategoryFacetSearchState;
+};
+
+export function buildCategoryFacetSearch(
+  engine: CommerceEngine,
+  props: CategoryFacetSearchProps
+): CategoryFacetSearch {
+  if (!loadCategoryFacetSearchReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const {showMoreResults, updateCaptions, ...restOfFacetSearch} =
+    buildCoreCategoryFacetSearch(engine, {
+      ...props,
+      executeFacetSearchActionCreator: executeCommerceFacetSearch,
+      executeFieldSuggestActionCreator: executeCommerceFieldSuggest,
+    });
+
+  return {
+    ...restOfFacetSearch,
+  };
```

Comment:
```suggestion
  return restOfFacetSearch;
```

Reply:
"

> 

Do we need to do a copy essentially, if no @Spuffynism suggestion. If yes, then keep it as is :D

Reply:
We don't need a copy, no!